CRAN pre-test WARNING

> Flavor: r-devel-linux-x86_64-debian-gcc
> Check: whether package can be installed, Result: WARNING
>   Found the following significant warnings:
>     libswe/sweph.c:7363:5: warning: 'strncpy' output may be truncated copying 256 bytes from a string of length 275 [-Wstringop-truncation]
>   See '/srv/hornik/tmp/CRAN/swephR.Rcheck/00install.out' for details.

--- a/external/patch/10_strncpy.diff
+++ b/external/patch/10_strncpy.diff
@@ -1,0 +1,15 @@
+--- external/src/sweph.c	2018-12-21 00:11:47.040930414 +0100
++++ src/libswe/sweph.c	2019-01-10 23:13:16.395002324 +0100
+@@ -7368,9 +7360,10 @@
+     }
+     // search string is traditional name
+     *sp = '\0';	/* cut off after first field to get star name, ',' -> '\0' */
+-    strncpy(fstar, s, SE_MAX_STNAME);
++    size_t length = strnlen(s, SE_MAX_STNAME);
++    memcpy(fstar, s, length);
++    fstar[length] = '\0';  /* force termination */
+     *sp = ',';  /* add comma again */
+-    fstar[SE_MAX_STNAME] = '\0';	/* force termination */
+     // remove white spaces from star name
+     while ((sp = strchr(fstar, ' ')) != NULL)
+       swi_strcpy(sp, sp+1);

--- a/src/libswe/sweph.c
+++ b/src/libswe/sweph.c
@@ -7362,9 +7362,8 @@ static int32 swi_fixstar_load_record(char *star, char *srecord, char *sname, cha
     *sp = '\0';	/* cut off after first field to get star name, ',' -> '\0' */
     size_t length = strnlen(s, SE_MAX_STNAME);
     memcpy(fstar, s, length);
-    fstar[length] = '\0';
+    fstar[length] = '\0';  /* force termination */
     *sp = ',';  /* add comma again */
-    fstar[SE_MAX_STNAME] = '\0';	/* force termination */
     // remove white spaces from star name
     while ((sp = strchr(fstar, ' ')) != NULL)
       swi_strcpy(sp, sp+1);

--- a/src/libswe/sweph.c
+++ b/src/libswe/sweph.c
@@ -7360,7 +7360,9 @@ static int32 swi_fixstar_load_record(char *star, char *srecord, char *sname, cha
     }
     // search string is traditional name
     *sp = '\0';	/* cut off after first field to get star name, ',' -> '\0' */
-    strncpy(fstar, s, SE_MAX_STNAME);
+    size_t length = strnlen(s, SE_MAX_STNAME);
+    memcpy(fstar, s, length);
+    fstar[length] = '\0';
     *sp = ',';  /* add comma again */
     fstar[SE_MAX_STNAME] = '\0';	/* force termination */
     // remove white spaces from star name

